aioxmpp/bookmarks/xso.py

```diff
@@ -224,8 +231,28 @@ class Storage(xso.XSO):
     .. attribute:: bookmarks
 
        A :class:`~xso.XSOList` of bookmarks.
-
     """
+
     TAG = (namespaces.xep0048, "storage")
 
     bookmarks = xso.ChildList([URL, Conference])
+
+
+def as_bookmark_class(xso_class):
+    """
+    Decorator to register `xso_class` as a custom bookmark class.
+
```

Comment:
Add a ``:raises TypeError: if `xso_class` is not a :class:`Bookmark` subclass``.